contracts/minter-suite/Minters/MinterDAExpSettlementV2.sol

```diff
@@ -0,0 +1,1296 @@
+        minterFilterAddress = _minterFilter;
+        minterFilter = IMinterFilterV0(_minterFilter);
+        require(
+            minterFilter.genArt721CoreAddress() == _genArt721Address,
+            "Illegal contract pairing"
+        );
+    }
+
+    /**
+     * @notice This function is intentionally not implemented for this version
+     * of the minter. Due to potential for unintended consequences, the
+     * function `manuallyLimitProjectMaxInvocations` should be used to manually
+     * and explicitly limit the maximum invocations for a project to a value
+     * other than the core contract's maximum invocations for a project.
+     * @param _projectId Project ID to set the maximum invocations for.
+     * @dev This function is included for interface conformance purposes only.
+     */
+    function setProjectMaxInvocations(
+        uint256 _projectId
+    ) external view onlyArtist(_projectId) {
+        revert("Not implemented");
+    }
+
+    /**
+     * @notice Manually sets the local maximum invocations of project `_projectId`
+     * with the provided `_maxInvocations`, checking that `_maxInvocations` is less
+     * than or equal to the value of project `_project_id`'s maximum invocations that is
+     * set on the core contract.
+     * @dev Note that a `_maxInvocations` of 0 can only be set if the current `invocations`
+     * value is also 0 and this would also set `maxHasBeenInvoked` to true, correctly short-circuiting
+     * this minter's purchase function, avoiding extra gas costs from the core contract's maxInvocations check.
+     * @param _projectId Project ID to set the maximum invocations for.
+     * @param _maxInvocations Maximum invocations to set for the project.
+     */
+    function manuallyLimitProjectMaxInvocations(
+        uint256 _projectId,
+        uint256 _maxInvocations
+    ) external onlyArtist(_projectId) {
+        // CHECKS
+        // ensure that the manually set maxInvocations is not greater than what is set on the core contract
+        uint256 coreInvocations;
+        uint256 coreMaxInvocations;
+        (
+            coreInvocations,
+            coreMaxInvocations
+        ) = _getProjectCoreInvocationsAndMaxInvocations(_projectId);
+        require(
+            _maxInvocations <= coreMaxInvocations,
+            "Cannot increase project max invocations above core contract set project max invocations"
+        );
+        require(
+            _maxInvocations >= coreInvocations,
+            "Cannot set project max invocations to less than current invocations"
+        );
+        // EFFECTS
+        ProjectConfig storage _projectConfig = projectConfig[_projectId];
+        // update storage with results
+        _projectConfig.maxInvocations = uint24(_maxInvocations);
+        // We need to ensure maxHasBeenInvoked is correctly set after manually setting the
+        // local maxInvocations value.
+        _projectConfig.maxHasBeenInvoked = coreInvocations == _maxInvocations;
```

Comment:
> This minter exposes a new gotcha that should be prevented by business practices. Put simply, once revenues are ready to be withdrawn, an artist should call withdrawArtistAndAdminRevenues. After a sellout, it is not recommended that the artist do anything with the local max invocations on the minter. If an auction were to sell out and an artist were to increase the local max invocations before withdrawing revenues, the most reasonable state could be interpreted to be that the auction should continue down to the base price, and the last price paid could drop. The result would be a lower settling price than expected. This would be an operational mistake by the artist, and would reduce the artist's revenues for that project while reducing the settling net-price paid by users.

Could we mitigate this even being a possible footgun, by making this line instead:

```
// Note: as a safety measure, never set a already true value to false, only allowing 
// `_maxInvocations`s functionally to be modified if have not already hit sell out
_projectConfig.maxHasBeenInvoked = (!_projectConfig.maxHasBeenInvoked) && coreInvocations == _maxInvocations;
```

Reply:
I believe this suggestion may have related implications for other invocation syncing methods throughout this minter, if this route seems like a good one in the interest of simplifying possible behavior.

Reply:
I quite like the concept of making `_projectConfig.maxHasBeenInvoked` a (almost, see last note below) 1-way switch.

To keep assumptions across the minter simpler, I have kept these lines of code the same, but added a require to this function that `_projectConfig.maxHasBeenInvoked` is false. Along with that require, also seems appropriate to require that the new `_maxInvocations` is gt `0`, which would forever prevent a project from using this minter (since it successful, it will immediately toggle the one-way switch) and be a nonsensical call by an artist.

Note that it is intentionally still possible for the local `maxHasBeenInvoked` to flip from true to false in the extreme edge case where the minter's local `maxHasBeenInvoked` is returning a false negative, which can only happen if max invocations were limited on the minter, the project used a different minter to go above this minter's local max invocations, then attempted to increase the local max invocations on this minter. This is not the footgun we are trying to protect against, and in that edge case, the artist's intentions cannot be easily predicted, so we can defer to whatever the artist calls.

Reply:
@jakerockland - These changes are implemented in [8475993](https://github.com/ArtBlocks/artblocks-contracts/pull/478/commits/8475993e1767f5e87535255ff3cb8e24dd6887b0) - do you mind resolving this if that looks good to you?

Reply:
Fantastic–LGTM ser.